src/winmain.cpp

```diff
@@ -1071,7 +1112,7 @@ int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE, PWSTR lpszCmdLine, int)
 				}
 
 				// install
-				std::wstring dlDest = _wgetenv(L"TEMP");
+				std::wstring dlDest = getDestDir();

```

Comment:
Because, we need this code twice: here (line 1115) and at the line 1279. Hence is the separate function.
Duplicating the same code is bad and could lead to other problems.

Also, usage of _wgetenv is discouraged since it's less secure:
https://stackoverflow.com/questions/48568707/getenv-function-may-be-unsafe-really

Reply:
I can try simplifying it if we only need to process TEMP, TMP and APPDATA.

Reply:
> Also, usage of `_wgetenv` is discouraged since it's less secure

Why is it not safe? The only explanation I can see is:
```
Returns a pointer to the environment table entry containing varname. It is not safe to modify the value of the environment variable using the returned pointer. Use the _putenv function to modify the value of an environment variable. 
```
(ref: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getenv-wgetenv?view=msvc-160)
If the string remains constant in WinGUP, there's no worry. So I don't see the advantage to use `_wgetenv_s`, even MS encourage it for their `*_s` promotion.

OTOH, according the description:
```
The return value is NULL if varname is not found in the environment table.
```

The code I suggested should be:
```
std::wstring dlDest;
wchar_t* pDlDestStr = _wgetenv(L"TEMP");
if (pDlDestStr)
    dlDest = pDlDestStr;
else
{
    pDlDestStr  = _wgetenv(L"TMP");
    if (pDlDestStr)
            dlDest = pDlDestStr;
    else
    {
        pDlDestStr  = _wgetenv(L"APPDATA");
        if (pDlDestStr)
        {
            dlDest = pDlDestStr;
            dlDest += L"Notepad++\";
            if (!PathFileExist(dlDest.c_str()))
                return -1; // or throw an exception
        }
        else
            return -1; // or throw an exception
    }
}
```

Reply:
> Because, we need this code twice: here (line 1115) and at the line 1279. Hence is the separate function.
Duplicating the same code is bad and could lead to other problems.

Then please remain the usage of `_wgetenv` and make the function return `wchar_t*`, and check the pointer's validity of returned value before using it.